src/mach/load_command.rs

```diff
@@ -1185,6 +1185,35 @@ pub struct EntryPointCommand {
 
 pub const SIZEOF_ENTRY_POINT_COMMAND: usize = 24;
 
+/// The build_version_command contains the min OS version on which this
+/// binary was built to run for its platform.  The list of known platforms and
+/// tool values following it.
+#[repr(C)]
+#[derive(Debug, Clone, Copy, Pread, Pwrite, IOread, IOwrite, SizeWith)]
+pub struct BuildVersionCommand {
+    /// LC_BUILD_VERSION
+    pub cmd: u32,
+    pub cmdsize: u32,
+    /// platform
+    pub platform: u32,
+    /// X.Y.Z is encoded in nibbles xxxx.yy.zz
+    pub minos: u32,
+    /// X.Y.Z is encoded in nibbles xxxx.yy.zz
+    pub sdk: u32,
+    /// number of tool entries following this
+    pub ntools: u32,
+}
+
+/// Build tool version
+#[repr(C)]
+#[derive(Debug, Clone, Copy, Pread, Pwrite, IOread, IOwrite, SizeWith)]
+pub struct BuildToolVersion {
```

Comment:
I can't remember how this part works, is the user responsible for extracting these out from `BuildVersionCommand` `ntools` field? Or do we normally provide a method on `BuildVersionCommand` to extract related substructs?

Reply:
I think the user is responsible for extracting `BuildToolVersion`, the `object` crate didn't provide any helper functions either.

* https://docs.rs/object/latest/object/macho/struct.BuildVersionCommand.html
* https://docs.rs/object/latest/object/macho/struct.BuildToolVersion.html